### PR TITLE
Updated the CloudEvent SDK, down-streaming those changes.

### PR DESCRIPTION
**Pull request by n3wscott**
Notable changes: HTTP casing, and new client.Option usage.

**Release Note**

```release-note
NONE
```

### CODE REVIEW

**Comment by n3wscott** [comment]
/assign @evankanderson 

since Ville is traveling

File: Gopkg.toml [hunk 1]
```diff
@@ -69,8 +69,6 @@ required = [
   name = "github.com/knative/serving"
   version = "v0.2.1"
 
-# STANZAS BELOW ARE GENERATED AND MAY BE WRITTEN - DO NOT MODIFY BELOW THIS LINE.
```
Comment (evankanderson) [inline Gopkg.toml]: ???

File: cmd/heartbeats/main.go [hunk 2]
```diff
@@ -110,7 +110,7 @@ func main() {
 			Data: hb,
 		}
 
-		if err := c.Send(event); err != nil {
+		if err := c.Send(context.TODO(), event); err != nil {
```
Comment (evankanderson) [inline cmd/heartbeats/main.go]: `context.Background()`.

File: Gopkg.toml (hunk 1 above)
Comment (n3wscott) [reply Gopkg.toml]: We do not auto-generate anymore. This line comes from kubebuilder

**Pull request merged**